oremda/pipeline/runner/rpc/client.py

```diff
@@ -0,0 +1,101 @@
+import uuid
+import asyncio
+from typing import Dict
+from fastapi_websocket_rpc import RpcMethodsBase, WebSocketRpcClient
+
+from oremda.typing import DisplayType, IdType, JSONType, PipelineJSON
+from oremda.pipeline.runner.context import GlobalContext
+from oremda.pipeline.messages import NotificationMessage, pipeline_created
+from oremda.pipeline.observer import ServerPipelineObserver
+from oremda.pipeline.displays import DisplayHandle1D, DisplayHandle2D
+from oremda.pipeline.models import PipelineModel, SerializablePipelineModel
+from oremda.pipeline import deserialize_pipeline
+from oremda.display import NoopDisplayHandle
+
+
+def unique_id():
+    return str(uuid.uuid4())
+
+
+class RpcClient(WebSocketRpcClient):
+    def __init__(self, uri: str, context: GlobalContext):
+        super().__init__(
+            uri, methods=PipelineRunnerMethods(context, self)  # type: ignore
+        )
+
+    async def notify_clients(self, message: JSONType, session_id: IdType):
+        await self.other.notify_clients(message=message, session_id=session_id)
+
+
+async def run_pipeline(session_id: IdType, pipeline_id: IdType, context: GlobalContext):
+    pipeline_ids = context.session_pipelines.get(session_id, set())
+    if pipeline_id not in pipeline_ids:
+        return
+
+    model = context.pipelines[pipeline_id]
+    pipeline = model.pipeline
+
+    # pipeline.run is a blocking function, run it in a separate thread to free the
+    # server to perform other tasks such as sending notifications
+    # TODO: convert pipeline.run to an async function
+    await asyncio.to_thread(pipeline.run)
```

Comment:
We can use [this decorator](https://github.com/OpenChemistry/oremda/blob/43951fa176ca8a6be19182c8f2ac12f3809691df/oremda/utils/asyncio.py#L7) on `pipeline.run` (or any blocking function) to make it an async function.